File: Civitas-Java/Civitas/src/main/java/civitas/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 * @author elena
 * @date 07/10/2019
 */

abstract class Sorpresa {
    protected String texto;
    protected int valor;
    private MazoSorpresas mazo;
    private Tablero tablero;
       
    Sorpresa(int valor, String texto){   //Resto de sorpresas
        this.mazo = null;
        this.tablero = null;
        this.texto = texto;
        this.valor = valor;
    }

    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){ return todos.size() > actual; }
    
    void informe(int actual, ArrayList<Jugador> todos){
        Diario diario = Diario.getInstance();
        diario.ocurreEvento("\nAplicando sorpresa, jugador " + todos.get(actual).getNombre());        
    }
    
    abstract void aplicarAJugador(int actual, ArrayList<Jugador> todos);
         
    @Override
    public String toString(){
        return "Sorpresa : " + this.texto + 
                "\nValor : " + this.valor;
    }
}

File: Civitas-Ruby/lib/sorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "diario"
require_relative "tipo_sorpresa"
require_relative "mazo_sorpresas"
require_relative "tablero"
require_relative "jugador"

module Civitas
  class Sorpresa    
    def initialize(valor, texto)
      @tablero = nil
      @mazo = nil
      @texto = texto
      @valor = valor
    end
    
    def informe(actual, todos)
      Diario.instance.ocurre_evento("Aplicando sorpresa a " + todos[actual].nombre)
    end
    
    private
    def aplicarajugador(actual, todos)     
    end
    
    public
    def jugadorcorrecto(actual, todos)
      actual < todos.length
    end
    
    def to_s
      'Sorpresa: ' + @texto.to_s + '\nValor: ' + @valor.to_s
    end
  end
end